src/core/linalg/src/dense/4C_linalg_fixedsizematrix.hpp

```diff
@@ -1922,6 +1922,20 @@ namespace Core::LinAlg
       for (unsigned int c = 1; c < i * j; ++c) result += *(++mat);
       return result / (i * j);
     }
+
+    template <class ValueType, unsigned int i, unsigned int j>
```

Comment:
instead of the `FOUR_C_ASSERT_ALWAYS`, you can write
```suggestion
    template <class ValueType, unsigned int i, unsigned int j> requires(i==1 || j == 1)
```